heron/metricscachemgr/tests/java/com/twitter/heron/metricscachemgr/metricscache/MetricsCacheQueryUtilsTest.java

```diff
@@ -0,0 +1,281 @@
+import com.twitter.heron.proto.system.Common;
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+
+import static com.twitter.heron.metricscachemgr.metricscache.MetricsCacheQueryUtils.toProtobuf;
+
+public class MetricsCacheQueryUtilsTest {
+  @Test
+  public void testFromProtoBufMetricInterval() {
+    TopologyMaster.MetricRequest request =
+        TopologyMaster.MetricRequest.newBuilder()
+            .setComponentName("c1")
+            .addInstanceId("i1").addInstanceId("i2")
+            .addMetric("m1").addMetric("m2")
+            .setInterval(100)// in seconds
+            .build();
+
+    MetricsCacheQueryUtils.MetricRequest request1 =
+        MetricsCacheQueryUtils.fromProtobuf(request);
+
+    Assert.assertEquals(1, request1.componentNameInstanceId.keySet().size());
+    Assert.assertEquals(true, request1.componentNameInstanceId.containsKey("c1"));
+    Assert.assertEquals(2, request1.componentNameInstanceId.get("c1").size());
+    Assert.assertEquals(true, request1.componentNameInstanceId.get("c1").contains("i1"));
+    Assert.assertEquals(true, request1.componentNameInstanceId.get("c1").contains("i2"));
+    Assert.assertEquals(100 * 1000, request1.endTime - request1.startTime); // in milli-seconds
+    Assert.assertEquals(MetricsCacheQueryUtils.Granularity.AGGREGATE_ALL_METRICS,
+        request1.aggregationGranularity);
+  }
+
+  @Test
+  public void testFromProtoBufMetricExplicitInterval() {
+    TopologyMaster.MetricRequest request =
+        TopologyMaster.MetricRequest.newBuilder()
+            .setComponentName("c1")
+            .addInstanceId("i1").addInstanceId("i2")
+            .addMetric("m1").addMetric("m2")
+            .setExplicitInterval(TopologyMaster.MetricInterval.newBuilder()
+                .setStart(100).setEnd(200)) // in seconds
+            .build();
+
+    MetricsCacheQueryUtils.MetricRequest request1 =
+        MetricsCacheQueryUtils.fromProtobuf(request);
+
+    Assert.assertEquals(1, request1.componentNameInstanceId.keySet().size());
+    Assert.assertEquals(true, request1.componentNameInstanceId.containsKey("c1"));
+    Assert.assertEquals(2, request1.componentNameInstanceId.get("c1").size());
+    Assert.assertEquals(true, request1.componentNameInstanceId.get("c1").contains("i1"));
+    Assert.assertEquals(true, request1.componentNameInstanceId.get("c1").contains("i2"));
+    Assert.assertEquals(2, request1.metricNames.size());
+    Assert.assertEquals(true, request1.metricNames.contains("m1"));
+    Assert.assertEquals(true, request1.metricNames.contains("m2"));
+    Assert.assertEquals(100 * 1000, request1.startTime); // in milli-seconds
+    Assert.assertEquals(200 * 1000, request1.endTime); // in milli-seconds
+    Assert.assertEquals(MetricsCacheQueryUtils.Granularity.AGGREGATE_ALL_METRICS,
+        request1.aggregationGranularity);
+  }
+
+  @Test
+  public void testFromProtoBufMetricMinutely() {
+    TopologyMaster.MetricRequest request =
```

Comment:
Can you initialize this in a setup method and have each test use the same request?